tests/components/jellyfin/__init__.py

```diff
@@ -1 +1,123 @@
 """Tests for the jellyfin integration."""
+
+from typing import Any
+from unittest.mock import MagicMock, Mock, patch
+
+from homeassistant.components.jellyfin.const import (
+    DATA_CLIENT,
+    DOMAIN,
+    MAX_STREAMING_BITRATE,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_URL, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+
+from .const import (
+    MOCK_AUTH_TOKEN,
+    MOCK_DEVICE_ID,
+    MOCK_SUCCESFUL_CONNECTION_STATE,
+    MOCK_SUCCESFUL_LOGIN_RESPONSE,
+    MOCK_TRACK,
+    MOCK_TRACK_ID,
+    MOCK_USER_ID,
+    TEST_PASSWORD,
+    TEST_URL,
+    TEST_USERNAME,
+)
+
+from tests.common import MockConfigEntry
+
+
+def create_mock_jellyfin_config_entry(hass: HomeAssistant) -> ConfigEntry:
+    """Add a test config entry."""
+    config_entry: MockConfigEntry = MockConfigEntry(
+        domain=DOMAIN,
+        data={
+            CONF_URL: TEST_URL,
+            CONF_USERNAME: TEST_USERNAME,
+            CONF_PASSWORD: TEST_PASSWORD,
+        },
+        title=f"{TEST_URL}",
+    )
+    config_entry.add_to_hass(hass)
+    return config_entry
+
+
+def _create_mock_jellyfin_api() -> Mock:
+    """Create a mock Jellyfin api."""
+    api = Mock()
+    api.audio_url = Mock(side_effect=_audio_url)
+    api.get_item = Mock(side_effect=_select_return_item)
+
+    return api
+
+
+def _select_return_item(item_id: str) -> dict[str, Any]:
+    """Return a mock item based on item id."""
+    if item_id == MOCK_TRACK_ID:
+        return MOCK_TRACK
+
+
+def _audio_url(item_id: str) -> str:
+    """Return streaming url based on item id."""
+    return f"{TEST_URL}/Audio/{item_id}/universal?UserId={MOCK_USER_ID}&DeviceId=Home+Assistant&api_key={MOCK_AUTH_TOKEN}&MaxStreamingBitrate={MAX_STREAMING_BITRATE}"
+
+
+def _create_mock_jellyfin_connection_manager(
+    connection_side_effect: Any = None, login_side_effect: Any = None
+) -> Mock:
+    """Return a mock Jellyfin connection manager."""
+    connection_manager = Mock()
+    if connection_side_effect:
```

Comment:
We want to avoid conditions in tests. It's better to parametrize the test with pytest.mark.parametrize or write different tests for different cases.

By using pytest fixtures we can compose the setup code for the tests as needed, normally. Eg the side effect could be a fixture that we can override per test case. Or just set the side effect on the mock in the test. We don't need a helper function for that.